### PR TITLE
Merge master into soroban soroban-xdr-next

### PR DESCRIPTION
**Pull request by 2opremio**
I needed this because of https://github.com/stellar/go/pull/4986 , but a merge was due anyways.

### CODE REVIEW

File: services/horizon/internal/ingest/main.go [hunk 1]
```diff
@@ -412,12 +412,6 @@ func (s *system) GetCurrentState() State {
 	return s.currentState
 }
 
-func (s *system) GetCurrentState() State {
```
[the remark this answers is not in the document]
Comment (2opremio) [reply services/horizon/internal/ingest/main.go:487]: The merge created two instances of the method. I corrected the merge.

File: ingest/ledgerbackend/toml.go [hunk 2]
```diff
@@ -70,12 +70,11 @@ type captiveCoreTomlValues struct {
 	BucketDirPath string `toml:"BUCKET_DIR_PATH,omitempty"`
 	// we cannot omitempty because 0 is a valid configuration for HTTP_PORT
 	// and the default is 11626
-	HTTPPort                  uint     `toml:"HTTP_PORT"`
-	PublicHTTPPort            bool     `toml:"PUBLIC_HTTP_PORT,omitempty"`
-	NodeNames                 []string `toml:"NODE_NAMES,omitempty"`
-	NetworkPassphrase         string   `toml:"NETWORK_PASSPHRASE,omitempty"`
-	PeerPort                  uint     `toml:"PEER_PORT,omitempty"`
-	LimitTxQueueSourceAccount bool     `toml:"LIMIT_TX_QUEUE_SOURCE_ACCOUNT,omitempty"`
```
[the remark this answers is not in the document]
Comment (2opremio) [reply ingest/ledgerbackend/toml.go]: It's further up in the struct

**Comment by 2opremio** [comment]
@tamirms I amended and force-pushed the merge. PTAL

**Comment by 2opremio** [comment]
Also, I changed the base branch. It was master and it should had been `soroban-xdr-next`

**Review by tsachiherman — APPROVED** [review]

**Comment by 2opremio** [comment]
Why did you merge? It contains a change i cherry-picked fir testing

**Comment by 2opremio** [comment]
@tsachiherman Reverting, since the commits where squashed instead of merged. Causing a million conflicts when merging master again.

**Pull request merged**